src/plugins/explore/public/components/results_summary/results_summary.tsx

```diff
@@ -40,7 +40,9 @@ export const ResultsSummary: React.FC<ResultsSummaryProps> = ({
   );
   const dataSetState = useSelector(selectDataset);
   const queryResults = useSelector(
-    (state: RootState) => state.results[state.query.query as string]?.hits?.hits
+    (state: RootState) =>
+      state.results[state.query.query as string]?.hits?.hits ||
+      Object.values(state.results)[0]?.hits?.hits // default query results
```

Comment:
Q: Why the cache key is `0`? shouldn't it be the default query?

Reply:
I think it is pretty much equivalent? Because in the initial state there will be only one result which is the first item of that list.

Reply:
I'm not sure how the current "cache" works, does it reset on every new search? I think it's better to use the current default query of "Logs" tab which is more robust.

Reply:
Done, now if it cannot retrieve the query result using the query string from the state, it will use the default query of that data source.